packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -1047,8 +1048,8 @@ abstract class State<T extends StatefulWidget> with Diagnosticable {
   /// configuration, the framework will update this property to refer to the new
   /// widget and then call [didUpdateWidget], passing the old configuration as
   /// an argument.
-  T get widget => _widget;
-  T _widget;
+  T get widget => _widget!;
+  T? _widget;
```

Comment:
we might get better diagnostics if we make _widget `late` rather than nullable

Reply:
this field is compared to null at some place, so it can't be late.

Reply:
ah, i see, yeah. We should probably refactor that a bit but let's do that in a separate PR.